### ISSUE TITLE
Excuse me, author, I have some questions.

### ISSUE THREAD

**Issue by CoconutLatte11** [JK-Liu7/AttentionMGT-DTA#17]
Please provide PDB_AF2 file

**Comment by LongLee220** [comment]
Hello, I found that, you can download PDB_AF2 files directly from the UniProt website using this URL: https://alphafold.ebi.ac.uk/files/AF-{pdb_id}-F1-model_v4.pdb. Just replace {pdb_id} with the specific protein ID.

Additionally, if you need to download multiple files in a batch, you can use the following Python script to automate the process. This script reads a list of protein IDs from a specified CSV file and downloads each corresponding file into a designated folder:

import os
import requests
import pandas as pd

def download_pdb_file(pdb_id, output_folder):
    url = f"https://alphafold.ebi.ac.uk/files/AF-{pdb_id}-F1-model_v4.pdb"
    output_file = os.path.join(output_folder, f"{pdb_id}.pdb")
    try:
        response = requests.get(url)
        if response.status_code == 200:
            with open(output_file, 'w') as file:
                file.write(response.text)
            print(f"Successfully downloaded {pdb_id}")
        else:
            print(f"Failed to download PDB file for {pdb_id}")
    except requests.exceptions.RequestException as e:
        print(f"Request failed for {pdb_id}: {e}")

output_folder = './data/Davis/PDB_AF2'
if not os.path exists(output_folder):
    os.makedirs(output_folder)

file_path = './data/Davis/Davis_protein_mapping.csv'
data = pd.read_csv(file_path)
protein_ids = data['PROTEIN_ID']

print(f"Starting download of {len(protein_ids)} PDB files...")
for pdb_id in protein_ids:
    download_pdb_file(pdb_id, output_folder)
print("Download process completed.")

**Comment by CoconutLatte11** [comment]
Thank you very much for your reply！~Are your processing results ideal compared with the paper?
Could you please provide us with your ESM-2 treatment method or results?

**Comment by LongLee220** [comment]
Not yet. If I have any new progress, I will send you a message. By the way, I noticed that there are no details about the dataset split. If you have any ideas, please let me know.

**Comment by CoconutLatte11** [comment]
hello，I didn't find it either！but，I found a way, hope it helps you.

import pandas as pd
from sklearn.model_selection import KFold
import os

# 定义数据路径和输出文件夹
file_path = 'input-data'
output_dir = 'data/Davis/DTA/fold/'
os.makedirs(output_dir, exist_ok=True)  # 创建顶层输出目录

# 读取数据集并进行数据检查
try:
    df = pd.read_csv(file_path)
    assert not df.empty, "数据集为空，请检查文件内容。"
except FileNotFoundError:
    raise FileNotFoundError(f"文件未找到，请检查路径：{file_path}")
except pd.errors.EmptyDataError:
    raise ValueError("CSV文件为空或格式不正确。")
except Exception as e:
    raise RuntimeError(f"读取数据集时出现错误：{e}")

# 如果需要，过滤掉有问题的数据（例如Q13153）
# df = df[df['PROTEIN_ID'] != 'Q13153']

# 初始化 KFold，设置为5折
kf = KFold(n_splits=5, shuffle=True, random_state=42)

# 分割并保存数据集
for i, (train_index, test_index) in enumerate(kf.split(df), start=1):
    train_df, test_df = df.iloc[train_index], df.iloc[test_index]

    # 为当前折创建文件夹
    fold_dir = os.path.join(output_dir, str(i))
    os.makedirs(fold_dir, exist_ok=True)

    # 保存当前折的训练集和测试集
    train_file = os.path.join(fold_dir, 'Davis_train.csv')
    test_file = os.path.join(fold_dir, 'Davis_test.csv')
    train_df.to_csv(train_file, index=False)
    test_df.to_csv(test_file, index=False)

    print(f'Fold {i} created with {len(train_df)} training samples and {len(test_df)} testing samples.')
    
print("数据预处理完成，所有折的文件已保存。")

**Comment by LongLee220** [comment]
Thank you very much, I also used this way, but the code still have some errors in the train_DTA.py

**Comment by angeload** [comment]
> Thank you very much, I also used this way, but the code still have some errors in the train_DTA.py

Hello @LongLee220. I've been trying to work with this code since it was launched, and while I've successfully generated the datasets, I'm having trouble running the train_DTA.py script. It's been quite frustrating not to have received a solution from the authors, as this seems to be a persistent issue. Like you, I'm eager to see this resolved, and I believe collaboration could be key to finding a solution. If you're interested, perhaps we can work together to tackle this problem.
